number_theory/prime.h

```diff
@@ -17,13 +17,59 @@
 namespace tql {
 namespace number_theory {
 
+// Sieve of Eratosthenes.
+// It decides whether a number is prime or not for the numbers up to given
+// (inclusive) limit.
+template <typename T>
+class Sieve {
+  static_assert(std::numeric_limits<T>::is_integer,
+                "Sieve must use integer types.");
+
+ public:
+  // The type of numbers used by the Sieve.
+  using type = T;
+
+  explicit Sieve(const T &num_limit)
```

Comment:
Shall we have comments about its time complexity?

Reply:
I thought of it, but in my opinion we should focus on what the given code does. We can have a nice documentation where we also mention complexities, show examples, mention caveats etc.